File: springmvc-xml-mybatis-maven/src/main/java/com/shiyan/service/impl/UserServiceImpl.java
package com.shiyan.service.impl;

import com.shiyan.dao.UserDao;
import com.shiyan.dto.UserDTO;
import com.shiyan.model.User;
import com.shiyan.service.UserService;
import org.springframework.stereotype.Service;

import javax.annotation.Resource;

/**
 * @author kevinfeng
 * @date 2019-05-25
 */
@Service
public class UserServiceImpl implements UserService {

    @Resource
    private UserDao userDao;

    @Override
    public UserDTO find(int id) {
        User user = userDao.findById(id);
        return convertModel2DTO(user);
    }

    private UserDTO convertModel2DTO(User user) {
        UserDTO userDTO = new UserDTO();
        userDTO.setId(user.getId());
        userDTO.setName(user.getName());
        userDTO.setAccount(user.getAccount());
        return userDTO;
    }
}

File: springboot-mybatis-gradle-kotlin/src/main/kotlin/com/shiyan/springboot/service/impl/UserServiceImpl.kt
package com.shiyan.springboot.service.impl

import com.github.pagehelper.PageHelper
import com.github.pagehelper.PageInfo
import com.shiyan.springboot.dao.UserMapper
import com.shiyan.springboot.model.User
import com.shiyan.springboot.service.UserService
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.stereotype.Service

/**
 *
 * @author kevinfeng
 * @date 2019-06-06
 */
@Service
class UserServiceImpl : UserService {

    @Autowired lateinit var userMapper: UserMapper

    override fun findAll(): List<User> {
        return userMapper.findAll()
    }

    override fun findUserById(id: Long): User {
        return userMapper.selectByPrimaryKey(id)
    }

    override fun page(pageNo: Int, pageSize: Int): PageInfo<User> {
        // 设置分页参数
        PageHelper.startPage<User>(pageNo, pageSize)
        val list = userMapper.findAll()
        // 使用PageInfo对象包装分页返回结果
        return PageInfo(list)
    }

}